### PR TITLE
[FLINK-19158][e2e] Fix wget timeout mechanism and cache config

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

Increase stability of java e2e tests

### CODE REVIEW

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java [hunk 1]
```diff
@@ -128,7 +128,11 @@ public Path getOrDownload(final String url, final Path targetDir) throws IOExcep
             Files.createDirectories(scopedDownloadDir);
             log.info("Downloading {}.", url);
             AutoClosableProcess.create(
-                            CommandLineWrapper.wget(url).targetDir(scopedDownloadDir).build())
+                            CommandLineWrapper.wget(url)
+                                    .targetDir(scopedDownloadDir)
+                                    .timeoutSecs(
+                                            downloadGlobalTimeout.getSeconds() / downloadMaxRetries)
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java:134]: shouldn't this use the `downloadAttemptTimeout`?

File: tools/azure-pipelines/jobs-template.yml [hunk 2]
```diff
@@ -199,7 +199,7 @@ jobs:
       condition: not(eq(variables['SKIP'], '1'))
     - task: Cache@2
       inputs:
-        key: e2e-cache | flink-end-to-end-tests/**/*.java, !**/avro/**
```
Comment (zentol) [inline tools/azure-pipelines/jobs-template.yml:202]: what is this for?

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/CommandLineWrapper.java [hunk 3]
```diff
@@ -45,6 +46,11 @@ public WGetBuilder targetDir(Path dir) {
             return this;
         }
 
+        public WGetBuilder timeoutSecs(long secs) {
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/CommandLineWrapper.java:49]: This should accept a Duration and internally convert it to whatever unit wget expects

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java (hunk 1 above)
Comment (rmetzger) [reply flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/cache/AbstractDownloadCache.java:134]: overlooked it, thx

File: tools/azure-pipelines/jobs-template.yml (hunk 2 above)
Comment (rmetzger) [reply tools/azure-pipelines/jobs-template.yml:202]: I believe the e2e cache never really worked, as it included only *.java files.

Comment (zentol) [reply tools/azure-pipelines/jobs-template.yml:202]: But isn't this line just a `key` to determine whether a cache should be used, and not an inclusion filter?

File: flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/CommandLineWrapper.java [hunk 4]
```diff
@@ -35,6 +36,7 @@ public static WGetBuilder wget(String url) {
 
         private final String url;
         private Path targetDir;
+        private int timeoutSecs;
```
Comment (zentol) [inline flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/CommandLineWrapper.java:39]: `Duration#getSeconds` returns a `long`; this currently fails compilation

File: tools/azure-pipelines/jobs-template.yml (hunk 2 above)
Comment (rmetzger) [reply tools/azure-pipelines/jobs-template.yml]: gosh, not sure what's going on with my brain. You are right.

**Review by zentol — APPROVED** [review]

**Comment by rmetzger** [comment]
Thanks for the review & approval.

I reported the build failure here: https://issues.apache.org/jira/browse/FLINK-21051

**Comment by rmetzger** [comment]
Merging now

**Pull request closed without merging**